[velbus] New functionality & Bug fix

Rosen01 opened the pull request:
<!--
Thanks for contributing to the openHAB project!
Please describe the goal and effect of your PR here.
Pay attention to the below notes and to *the guidelines* for this repository.
Feel free to delete any comment sections in the template (starting with "<!--").

ATTENTION: Don't use "git merge" when working with your pull request branch!
This can clutter your Git history and make your PR unusable.
Use "git rebase" instead. See this forum post for further details:
https://community.openhab.org/t/rebase-your-code-or-how-to-fix-your-git-history-before-requesting-a-pull/129358

All PRs should be created using the "main" branch as base.
Important bugfixes are cherry-picked by maintainers to the patch release branch after a PR has been merged.

Add one or more appropriate labels to make your PR show up in the release notes.
E.g. enhancement, bug, documentation, new binding
This can only be done by yourself if you already contributed to this repo.

If your PR's code is not backward compatible with previous releases (which
should be avoided), add a message to the release notes by filing another PR:
https://github.com/openhab/openhab-distro/blob/main/distributions/openhab/src/main/resources/bin/update.lst

# Title

Provide a short summary in the *Title* above. It will show up in the release notes.
For example:
- [homematic] Improve communication with weak signal devices
- [timemachine][WIP] Initial contribution

# Description

Please give a few sentences describing the overall goals of the pull request.
Give enough details to make the improvement and changes of the PR understandable
to both developers and tech-savy users.

Please keep the following in mind:
- What is the classification of the PR, e.g. Bugfix, Improvement, Novel Addition, ... ?
- Did you describe the PRs motivation and goal?
- Did you provide a link to any prior discussion, e.g. an issue or community forum thread?
- Did you describe new features for the end user?
- Did you describe any noteworthy changes in usage for the end user?
- Was the documentation updated accordingly, e.g. the add-on README?
- Does your contribution follow the coding guidelines:
  https://www.openhab.org/docs/developer/guidelines.html
- Did you check for any (relevant) issues from the static code analysis:
  https://www.openhab.org/docs/developer/bindings/#include-the-binding-in-the-build
- Did you sign-off your work:
  https://www.openhab.org/docs/developer/contributing.html#sign-your-work

# Testing

Your pull request will automatically be built and available under the following folder:
https://openhab.jfrog.io/ui/native/libs-pullrequest-local/org/openhab/addons/bundles/

It is a good practice to add a URL to your built JAR in this pull request description,
so it is easier for the community to test your Add-on.
If your pull request contains a new binding, it will likely take some time
before it is reviewed and processed by maintainers.
That said, consider submitting your Add-on in the Marketplace:
https://community.openhab.org/c/marketplace/69

Don't forget to submit a thread about your Add-on in the openHAB community:
https://community.openhab.org/c/add-ons 
-->

New functionality : 
- Add modules : VMB4LEDPWM-20, VMB8DC-20, VMBPIRO-10, VMBPIR-20, VMB8IN-20

Fix bug : 
- Correct typo in thing type for module VMB6PB-20.
- Remove properties 'Refresh Interval' from modules VMBPIRC, VMBPIRM.
- Add CH8 properties on VMB7IN module.
- Channels name retrieved from modules not set correctly to the right channel.
- Remove property 'Refresh Interval' from modules VMBPIRC, VMBPIRM.
- Remove sub-address property for modules VMBPIRO.
- VMBDALI, VMBDALI-20 use an exponential curve system for the dimming.
- Update readme.

Link for testing : [Releases](https://github.com/Rosen01/openhab-addons/releases)

Signed-off-by: Daniel Rosengarten <github@praetorians.be>

lsiepel commented:
Thanks for this PR! Before i review i like to ask you if it would be possible to split this PR in 2 (or more) seperate PR's. One that fixes bugs and one that adds features (channels).
If not we have to deal with it ;-) but in general we prefer smaller PR's as they are easier to review, backport and show up independent in the release notes.

Rosen01 commented:
Hi @lsiepel,

Sorry but I'm not familiar enough with GitHub to manage the split off the changes in different pull requests.
At the beginning, I just added new modules, but during the tests we found some old bugs in the code that I've directly corrected.

I know, it seems there are a lot of changes, but if you take a look : 
I did some formatting of the code to be more "compliant" (changing the 1st letter from uppercase to lowercase of the functions). 
I also split the .xml files from config, thing and update directories into smaller files and with more accurate names.

lsiepel reviewed (approved):
Thanks, LGTM.
Hope to get a confirmation from @cedricboon before i merge this.

For future PR's please split bug fixes from enhancements. This can be done in many ways, but for instance by creating an enhancement brancht on top of the bug fix branch. Or two completely independend branches. This would be easier to review, backport, list in release notes and fix regressions when needed. Thank you!
